analyzer/protocol/ldap/asn1.spicy

```diff
@@ -129,19 +129,18 @@ type ASN1OctetString = unit(len: uint64, constructed: bool) {
 # https://www.obj-sys.com/asn1tutorial/node124.html
 
 type ASN1String = unit(tag: ASN1Tag, len: uint64) {
-  var value: string = "";
+  var encoding: hilti::Charset;
 
-  : ASN1OctetString(len, tag.constructed) {
+  on %init {
     switch ( tag.type_ ) {
-
       # see "Restricted Character String Types" in
       # "Generic String Encoding Rules (GSER) for ASN.1 Types"
       # (https://datatracker.ietf.org/doc/html/rfc3641#section-3.2)
 
       case ASN1Type::PrintableString,
            ASN1Type::GeneralizedTime,
            ASN1Type::UTCTime: {
-        self.value = $$.value.decode(hilti::Charset::ASCII);
```

Comment:
Might be more convenient to keep the decoding, so that the user of the type doesn't need to worry about that charset anymore.

Reply:
We now decode this after extracting `value`, and convert the whole unit to a string with a unit convert. The user should never need to worry how `ASN1String`s are encoded.

Reply:
Ah, I see it now, nevermind.